server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -623,7 +698,43 @@ public int compare(File o1, File o2)
           if (Ints.tryParse(segmentDir.getName()) == null) {
             continue;
           }
-
+          QueryableIndex queryableIndex = null;
+          try {
+            queryableIndex = IndexIO.loadIndex(segmentDir);
```

Comment:
I dont think it can be closed here as a reference to it stored in the sink. This PR did not really change anything wrt to loading the index (see  https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java#L637 ) . I'm not sure where this will gets closed (probably when FireHydrant.swapSegment(..)) happens.  If it does not get closed then that is probably an existing bug and should be handled in a separate  PR.
@gianm do you know when this will get closed?